### PR TITLE
docs(registry): Added documentation on gulp.registry

### PR DESCRIPTION
**Pull request by SyntaxRules**
Adresses https://github.com/gulpjs/gulp/issues/1003

No code changes. No breaking changes.

### CODE REVIEW

**Comment by SyntaxRules** [comment]
Added more examples for the discussion in #1003 (After I checked them for syntax errors and ran them locally)

File: docs/API.md [hunk 1]
```diff
@@ -653,6 +653,82 @@ gulp.tree({ deep: true })
 */
 ```
 
+### gulp.registry([registry])
+
+Get or set the current registry. Inherited from [undertaker] see the undertaker documention on [registries](https://github.com/phated/undertaker#registryregistryinstance). Using this you can import registries from other sources/files. Importing your own registry has at least three use cases:
```
Comment (callumacrae) [inline docs/API.md]: needs a comma or semi-colon or something

File: docs/API.md [hunk 2]
```diff
@@ -653,6 +653,82 @@ gulp.tree({ deep: true })
...
+
+- [Sharing tasks](https://github.com/phated/undertaker#sharing-tasks)
+- [Sharing functionality](https://github.com/phated/undertaker#sharing-functionalities). (ex. you could override the task prototype to add some additional logging, or include some config settings.)
```
Comment (callumacrae) [inline docs/API.md]: e.g. is latin for "for example"

File: docs/API.md [hunk 3]
```diff
@@ -653,6 +653,82 @@ gulp.tree({ deep: true })
...
+- Handling other behavior that hooks into the registry lifecycle (see [gulp-hub](https://github.com/frankwallis/gulp-hub) for an example)
```
Comment (callumacrae) [inline docs/API.md]: This doesn't mean anything to someone who doesn't already know what it means

File: docs/API.md [hunk 4]
```diff
@@ -653,6 +653,82 @@ gulp.tree({ deep: true })
...
+
+```js
+//gulpfile.js
+var gulp = require('gulp');
+
+var companyTasks = require('./myCompanyTasksRegistry.js');
+
+gulp.registry(companyTasks);
+
+gulp.task('one', gulp.parallel('someCompanyTask', function(done) {
```
Comment (callumacrae) [inline docs/API.md]: Could you make these examples a bit more real life?

File: docs/API.md (hunk 1 above)
Comment (SyntaxRules) [inline docs/API.md]: Added

File: docs/API.md (hunk 2 above)
Comment (SyntaxRules) [inline docs/API.md]: Changed

File: docs/API.md (hunk 3 above)
Comment (SyntaxRules) [inline docs/API.md]: I'm struggling with how to word this better. Any Suggestions?

**Comment by contra** [comment]
@callumacrae Look good with updates?

**Comment by phated** [comment]
I'd like to review this also.  Hoping to have more time coming up

**Comment by callumacrae** [comment]
Grammar lgtm :+1:

**Comment by phated** [comment]
Rebasing caused this to be out of sync but I was able to cherry pick the commits onto the 4.0 branch.  I trimmed some extra stuff I didn't think was needed but it did outline some underlying problems in undertaker that I need to solve.  Thanks!

**Pull request closed without merging**